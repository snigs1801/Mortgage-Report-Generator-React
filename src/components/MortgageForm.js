import React, { useState, Fragment } from "react";
import AmountInput from "./AmountInput";
import PercentageInput from "./PercentageInput";
import AmotizationOption from "./AmotizationOption";
import PaymentSummary from "./PaymentSummary";
import { calculateMonthlyPayment } from "../utils/utils.js";

const MortgageForm = () => {

  // set initial state values (react hooks methods used)

  const [mortgageAmount, setMortgageAmount] = useState('');
  const [interestRate, setInterestRate] = useState('');
  const [downPayment, setDownPayment] = useState('');
  const [mortgageTerm, setMortgageTerm] = useState('');
  const [showSummary, setShowSummary] = useState(false);
  const [monthlyPayment, setMonthlyPayment] = useState(0);

  // enables and disables the calculate button 
  const ifSubmitDisabled =
    mortgageAmount === '' ||
    mortgageTerm === '' ||
    downPayment === '' ||
    interestRate === ''
  ;

  // handles the calculate button click
  const calculate = e => {
    e.preventDefault();
    setMonthlyPayment(
      calculateMonthlyPayment(
        mortgageAmount,
        interestRate,
        downPayment,
        mortgageTerm
      )
    );

    // show summary set is set to 'true'
    setShowSummary(true);
  };

  // returns the entire Mortgage Form
  // Used React Fragment Pattern to return multiple elements
  // The payment summary is 'hidden' by default and is visible when form is submitted

  return (
    <Fragment>
      <form className="form-group" onSubmit={calculate}>
        <AmountInput
          labelName="Mortgage Amount"
          targetName="mortgageAmount"
          targetValue={mortgageAmount}
          setMortgageAmount={setMortgageAmount}
        />
        <PercentageInput
          labelName="Interest Rate"
          targetName="interestRate"
          targetValue={interestRate}
          setInterestRate={setInterestRate}
        />
        <div className="input-group">
          <label>Down Payment</label>
          <div className="input-box">
            <input
              type="number"
              name="downPayment"
              className="right-flat"
              value={downPayment}
              placeholder="DownPayment Rate"
              onChange={e => setDownPayment(e.target.value)}
            />
            <div className="add-on left-align">%</div>
          </div>
        </div>
        <div className="input-group">
          <label>Amortization Period</label>
          <div className="radio-group">
            <AmotizationOption
              mortgageTerm={mortgageTerm}
              targetValue="5"
              setMortgageTerm={setMortgageTerm}
            />
            <AmotizationOption
              mortgageTerm={mortgageTerm}
              targetValue="10"
              setMortgageTerm={setMortgageTerm}
            />
            <AmotizationOption
              mortgageTerm={mortgageTerm}
              targetValue="15"
              setMortgageTerm={setMortgageTerm}
            />
            <AmotizationOption
              mortgageTerm={mortgageTerm}
              targetValue="20"
              setMortgageTerm={setMortgageTerm}
            />
            <AmotizationOption
              mortgageTerm={mortgageTerm}
              targetValue="25"
              setMortgageTerm={setMortgageTerm}
            />
          </div>
        </div>
        <input
          type="submit"
          value="Calculate"
          className="calculate-btn"
          disabled={ifSubmitDisabled}
        />
      </form>

      <PaymentSummary
        show={showSummary}
        mortgageAmount={mortgageAmount}
        interestRate={interestRate}
        downPayment={downPayment}
        mortgageTerm={mortgageTerm}
        monthlyPayment={monthlyPayment}
        setShowSummary={setShowSummary}
      />
    </Fragment>
  );
};

export default MortgageForm;
