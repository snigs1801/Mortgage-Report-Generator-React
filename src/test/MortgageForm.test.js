
import React from 'react';import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
 
import MortgageForm from '../components/MortgageForm';

Enzyme.configure({ adapter: new Adapter() });

describe('MortgageForm component Test Cases', () => {
  const container = shallow(<MortgageForm />);
  
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have a submit button', () => { 
    expect(container.find('input[type="submit"]').length).toEqual(1);
  });
  
  it('should have proper props for submit button', () => { 
    expect(container.find('input[type="submit"]').props()).toEqual({
      className: 'calculate-btn',
      type: 'submit',
      value: "Calculate",
      disabled: true
  }); 
  
  });
});