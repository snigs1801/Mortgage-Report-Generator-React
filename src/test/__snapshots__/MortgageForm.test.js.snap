// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MortgageForm component Test Cases should match the snapshot 1`] = `"<form class=\\"form-group\\"><div class=\\"input-group\\"><label>Mortgage Amount</label><div class=\\"input-box\\"><div class=\\"add-on right-align\\">CAD</div><input type=\\"text\\" name=\\"mortgageAmount\\" class=\\"left-align\\" value=\\"\\" placeholder=\\"Mortgae Amount\\"/></div></div><div class=\\"input-group\\"><label>Interest Rate</label><div class=\\"input-box\\"><input type=\\"number\\" class=\\"right-align\\" name=\\"interestRate\\" value=\\"\\" placeholder=\\"Mortgage Rate\\"/><div class=\\"add-on left-align\\">%</div></div></div><div class=\\"input-group\\"><label>Down Payment</label><div class=\\"input-box\\"><input type=\\"number\\" name=\\"downPayment\\" class=\\"right-flat\\" value=\\"\\" placeholder=\\"DownPayment Rate\\"/><div class=\\"add-on left-align\\">%</div></div></div><div class=\\"input-group\\"><label>Amortization Period</label><div class=\\"radio-group\\"><label><input type=\\"radio\\" name=\\"mortgageTerm\\" value=\\"5\\"/>5</label><label><input type=\\"radio\\" name=\\"mortgageTerm\\" value=\\"10\\"/>10</label><label><input type=\\"radio\\" name=\\"mortgageTerm\\" value=\\"15\\"/>15</label><label><input type=\\"radio\\" name=\\"mortgageTerm\\" value=\\"20\\"/>20</label><label><input type=\\"radio\\" name=\\"mortgageTerm\\" value=\\"25\\"/>25</label></div></div><input type=\\"submit\\" value=\\"Calculate\\" class=\\"calculate-btn\\" disabled=\\"\\"/></form><div class=\\"summaryHide\\"><h2>Your Mortgage Report</h2><div class=\\"results\\"><div class=\\"resultRow\\"><div class=\\"resultLabel\\">Mortgage Amount</div><div class=\\"resultValue\\"></div></div></div></div>"`;
